pages/overview/visualizations/issue_assignment.py

```diff
@@ -0,0 +1,265 @@
+    df_assign = dates.to_frame(index=False, name="start_date")
+
+    # offset end date column by interval
+    if interval == "D":
+        df_assign["end_date"] = df_assign.start_date + pd.DateOffset(days=1)
+    elif interval == "W":
+        df_assign["end_date"] = df_assign.start_date + pd.DateOffset(weeks=1)
+    elif interval == "M":
+        df_assign["end_date"] = df_assign.start_date + pd.DateOffset(months=1)
+    else:
+        df_assign["end_date"] = df_assign.start_date + pd.DateOffset(years=1)
+
+    # dynamically apply the function to all dates defined in the date_range to create df_status
+    df_assign["Assigned"], df_assign["Unassigned"] = zip(
+        *df_assign.apply(
+            lambda row: issue_assignment(df, row.start_date, row.end_date),
+            axis=1,
+        )
+    )
+
+    # formatting for graph generation
+    if interval == "M":
+        df_assign["start_date"] = df_assign["start_date"].dt.strftime("%Y-%m")
+    elif interval == "Y":
+        df_assign["start_date"] = df_assign["start_date"].dt.year
+
+    return df_assign
+
+
+def create_figure(df: pd.DataFrame, interval):
+    # time values for graph
+    x_r, x_name, hover, period = get_graph_time_values(interval)
+
+    # making a line graph if the bin-size is small enough.
+    if interval == "D":
+        fig = go.Figure(
+            [
+                go.Scatter(
+                    name="Assigned",
+                    x=df["start_date"],
+                    y=df["Assigned"],
+                    mode="lines",
+                    showlegend=True,
+                    hovertemplate="Issues Assigned: %{y}<br>%{x|%b %d, %Y} <extra></extra>",
+                    marker=dict(color=color_seq[0]),
+                ),
+                go.Scatter(
+                    name="Unassigned",
+                    x=df["start_date"],
+                    y=df["Unassigned"],
+                    mode="lines",
+                    showlegend=True,
+                    hovertemplate="Issues Unassigned: %{y}<br>%{x|%b %d, %Y}<extra></extra>",
+                    marker=dict(color=color_seq[3]),
+                ),
+            ]
+        )
+    else:
+        fig = px.bar(
+            df,
+            x="start_date",
+            y=["Assigned", "Unassigned"],
+            color_discrete_sequence=[color_seq[0], color_seq[3]],
+        )
+
+        # edit hover values
+        fig.update_traces(hovertemplate=hover + "<br>Issues: %{y}<br><extra></extra>")
+
+        fig.update_xaxes(
+            showgrid=True,
+            ticklabelmode="period",
+            dtick=period,
+            rangeslider_yaxis_rangemode="match",
+            range=x_r,
+        )
+
+    # layout specifics for both styles of plots
+    fig.update_layout(
+        xaxis_title="Time",
+        yaxis_title="Issues",
+        legend_title="Types",
+        font=dict(size=14),
+    )
+
+    return fig
+
+
+def issue_assignment(df, start_date, end_date):
```

Comment:
docstring.

It's difficult to understand what's being counted in this function. Grouping operations and describing the data in comments is nice to have when reading code like this since we can't see the data without print statements.